web/js/containers/sidebar/layer-row.js

```diff
@@ -96,11 +96,13 @@ function LayerRow (props) {
           isMobile={isMobile}
         />
       );
-    } if (!isLoading) {
-      requestPalette(layer.id);
     }
   };
 
+  useEffect(() => {
+    if (!isLoading && layer && hasPalette) requestPalette(layer.id);
+  }, [layer]);
```

Comment:
Not positive, but maybe you want to use `[layer.id])`. Uncertain how that object would be compared in this case.

Reply:
I _think_ it's an object reference comparison and since the `layer` prop shouldn't be changing on this component I think it will have the same effect if we use `layer.id`

Reply:
Actually, after some debugging on this it looks like this runs fewer times when we check for `layer.id` instead of `layer` so that seems like the better approach.  Also it looks like a few other places in the app are throwing this warning so I want to see if I can track those down and fix them in the same PR.